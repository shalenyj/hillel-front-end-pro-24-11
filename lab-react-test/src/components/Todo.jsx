export default function Todo({ todo }){
  const text = todo.completed ? <strike>{todo.title}</strike> : todo.title
  return (
    <div data-testid={`todo-${todo.id}`} >{text}</div>
  )
}