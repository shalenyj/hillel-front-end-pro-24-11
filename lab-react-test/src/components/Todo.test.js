import { render, screen } from "@testing-library/react";
import renderer from 'react-test-renderer'

import Todo from "./Todo";

describe('components: Todo', () => {
  const todo = { id: 1, title:'all wash car', completed: false}
  it('should be in DOM', async() => {
    render(<Todo todo={todo} />);
    const element = await screen.findByTestId('todo-1')
    expect(element).toBeInTheDocument()
  })
  it('should have text from todo.title', async() => {
    render(<Todo todo={todo} />)
    const element = await screen.findByTestId('todo-1')
    expect(element).toHaveTextContent(todo.title)
  })
  it('should have strike tag if completed', async() => {
    const todo = { id: 2, title:'all wash dishes', completed: true }
    render(<Todo todo={todo} />)
    const element = await screen.findByTestId('todo-2')
    expect(element).toContainHTML(`<strike>${todo.title}</strike>`)
  })
  it('should have strike tag if not completed', async() => {
    const todo = { id: 3, title:'wash desk', completed: false }
    render(<Todo todo={todo} />)
    const element = await screen.findByTestId('todo-3')
    expect(element).not.toContainHTML(`<strike>${todo.title}</strike>`)
  })

  it('should match snapshot', () => {
    const todo = { id: 3, title:'wash desk', completed: false }
    const element = renderer.create(<Todo todo={todo} />)

    const elementAsJson = element.toJSON();
    expect(elementAsJson).toMatchSnapshot()
  })
  it('should match snapshot', () => {
    const todo = { id: 3, title:'wash desk', completed: true }
    const element = renderer.create(<Todo todo={todo} />)

    const elementAsJson = element.toJSON();
    expect(elementAsJson).toMatchSnapshot()
  })
})