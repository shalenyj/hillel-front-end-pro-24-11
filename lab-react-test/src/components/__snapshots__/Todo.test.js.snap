// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components: Todo should match snapshot 1`] = `
<div
  data-testid="todo-3"
>
  wash desk
</div>
`;

exports[`components: Todo should match snapshot 2`] = `
<div
  data-testid="todo-3"
>
  <strike>
    wash desk
  </strike>
</div>
`;
